examples/prodlda.py

```diff
@@ -0,0 +1,207 @@
+        logtheta_scale = jnp.exp(0.5 * logtheta_logvar)
+        return logtheta_loc, logtheta_scale
+
+
+class HaikuDecoder:
+    def __init__(self, vocab_size, dropout_rate):
+        self._dropout_rate = dropout_rate
+        self._vocab_size = vocab_size
+
+    def __call__(self, inputs, is_training):
+        dropout_rate = self._dropout_rate if is_training else 0.0
+        h = hk.dropout(hk.next_rng_key(), dropout_rate, inputs)
+        h = hk.Linear(self._vocab_size)(h)
+        h = hk.BatchNorm(create_scale=True, create_offset=True, decay_rate=0.1)(
+            h, is_training
+        )
+        return jax.nn.softmax(h)
+
+
+def model(docs, hyperparams, is_training=False):
+    decoder = haiku_module(
+        "decoder",
+        # use `transform_with_state` for BatchNorm
+        hk.transform_with_state(
+            HaikuDecoder(hyperparams["vocab_size"], hyperparams["dropout_rate"])
+        ),
+        input_shape=(1, hyperparams["num_topics"]),
+        apply_rng=True,
+        # to ensure proper initialisation of BatchNorm we must
+        # initialise with is_training=True
+        is_training=True,
+    )
+
+    with numpyro.plate(
+        "documents", docs.shape[0], subsample_size=hyperparams["batch_size"]
+    ):
+        batch_docs = numpyro.subsample(docs, event_dim=1)
+        logtheta_loc = jnp.zeros((batch_docs.shape[0], hyperparams["num_topics"]))
+        logtheta_scale = jnp.ones((batch_docs.shape[0], hyperparams["num_topics"]))
+        logtheta = numpyro.sample(
+            "logtheta", dist.Normal(logtheta_loc, logtheta_scale).to_event(1)
+        )
+        theta = jax.nn.softmax(logtheta)
+
+        count_param = decoder(numpyro.prng_key(), theta, is_training)
+
+        total_count = batch_docs.sum(-1).max().astype(int)
+        numpyro.sample(
+            "obs", dist.Multinomial(total_count, count_param), obs=batch_docs
+        )
+
+
+def guide(docs, hyperparams, is_training=False):
+    encoder = haiku_module(
+        "encoder",
+        # use `transform_with_state` for BatchNorm
+        hk.transform_with_state(
+            HaikuEncoder(
+                hyperparams["vocab_size"],
+                hyperparams["num_topics"],
+                hyperparams["hidden"],
+                hyperparams["dropout_rate"],
+            )
+        ),
+        input_shape=(1, hyperparams["vocab_size"]),
+        apply_rng=True,
+        # to ensure proper initialisation of BatchNorm we must
+        # initialise with is_training=True
+        is_training=True,
+    )
+    with numpyro.plate(
+        "documents", docs.shape[0], subsample_size=hyperparams["batch_size"]
+    ):
+        batch_docs = numpyro.subsample(docs, event_dim=1)
+        logtheta_loc, logtheta_scale = encoder(
+            numpyro.prng_key(), batch_docs, is_training
+        )
+        numpyro.sample(
+            "logtheta", dist.Normal(logtheta_loc, logtheta_scale).to_event(1)
+        )
+
+
+def plot_word_cloud(b, ax, n):
```

Comment:
the arguments are `plot_word_cloud(b, ax, vocab, n)`

Reply:
Oops, good catch!